cassandra-dac/src/test/java/org/sunbird/cassandra/CassandraMockTest.java

```diff
@@ -0,0 +1,643 @@
+    Response response =
+        operation.getPropertiesValueById(
+            cassandraKeySpace, "address", "123", JsonKey.ID, JsonKey.CITY, JsonKey.ADD_TYPE);
+    assertTrue(response.getResult().size() > 0);
+  }
+
+  @Test
+  public void testCassandraGetPropertiesValueByIdFailedOperation() throws Exception {
+
+    Throwable exception = null;
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenThrow(
+            new ProjectCommonException(
+                ResponseCode.SERVER_ERROR.getErrorCode(),
+                ResponseCode.SERVER_ERROR.getErrorMessage(),
+                ResponseCode.SERVER_ERROR.getResponseCode()));
+
+    try {
+      operation.getPropertiesValueById(
+          cassandraKeySpace, "address", "123", JsonKey.ID, JsonKey.CITY, JsonKey.ADD_TYPE);
+    } catch (Exception ex) {
+      exception = ex;
+    }
+    assertTrue((((ProjectCommonException) exception).getResponseCode()) == 500);
+  }
+
+  @Test
+  public void testCassandraGetRecordByIdOperation() {
+
+    List<Row> rows = new ArrayList<>();
+    Row row = Mockito.mock(Row.class);
+    rows.add(row);
+    when(resultSet.all()).thenReturn(rows);
+
+    when(session.execute(boundStatement.bind("123"))).thenReturn(resultSet);
+    when(session.execute(where)).thenReturn(resultSet);
+    when(selectBuilder.from(Mockito.anyString(), Mockito.anyString())).thenReturn(selectQuery);
+    when(selectSelection.all()).thenReturn(selectBuilder);
+
+    Response response = operation.getRecordById(cassandraKeySpace, "address", "123");
+    assertTrue(response.getResult().size() > 0);
+  }
+
+  @Test
+  public void testCassandraGetRecordByIdFailedOperation() throws Exception {
+
+    Throwable exception = null;
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenThrow(
+            new ProjectCommonException(
+                ResponseCode.SERVER_ERROR.getErrorCode(),
+                ResponseCode.SERVER_ERROR.getErrorMessage(),
+                ResponseCode.SERVER_ERROR.getResponseCode()));
+
+    try {
+      operation.getRecordById(cassandraKeySpace, "address", "123");
+    } catch (Exception ex) {
+      exception = ex;
+    }
+    assertTrue((((ProjectCommonException) exception).getResponseCode()) == 500);
+  }
+
+  @Test
+  public void testCassandraGetRecordByPropertiesOperation() throws Exception {
+
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.USER_ID, "USR1");
+    map.put(JsonKey.ADD_TYPE, "addrType");
+
+    List<Row> rows = new ArrayList<>();
+    Row row = Mockito.mock(Row.class);
+    rows.add(row);
+    when(resultSet.all()).thenReturn(rows);
+
+    Response response = operation.getRecordsByProperties(cassandraKeySpace, "address", map);
+    assertTrue(response.getResult().size() > 0);
+  }
+
+  @Test
+  public void testCassandraGetRecordByPropertiesFailedOperation() throws Exception {
```

Comment:
why test-case expects with exact same data to fail once, and pass once?

what is the logic?